### DISCUSSION TITLE
Applying practical DDD in Agile (Scrum) teams at scale

### DISCUSSION THREAD

**Discussion by geovanneb** [ddd-crew/ddd-starter-modelling-process#67] [Show and tell] [answered]
Hi DDD Crew,

We have been applying your "[DDD Starter Modelling Process](https://github.com/ddd-crew/ddd-starter-modelling-process)" at scale in a huge business domain involving over ~25 Agile teams. 

Throughout this journey, we have gained some insights and learnings, and I started thinking about how to contribute back to the community based on these learnings.

I recently spoke with @Max-Git , and we agreed that it would be a good idea to share an article detailing our experiences. 

I have created the article [here](https://github.com/geovanneb/ddd-at-scale). 

Would you mind reviewing it to see if it might be helpful for your DDD crew group?

I understand that your guide is not a rigid framework, but we have been applying it in our context in a very flexible way, and it has been really helpful to us. 

Any feedback is welcome.

Kind regards,

**Comment by Max-Git** [comment]
Hey @geovanneb 

Thanks for taking the time to write this article. I think it will definitely help Scrum teams giving them ideas on how to implement DDD. 

Here are my remarks:
Typo
« In Agile, they are _appear_ as Business and Technical Refinements, User Stories, Sprint Reviews, User Acceptance Testing, and other practices. »

Champions: I would avoid this term as it sounds a bit similar to Hero. You might think that you have people who know and those who don’t. I would use terms like Leaders or Coaches instead.
You could also think of them as members of Enabling Teams (as described in Team Topologies)

Book: « reading a good book like the blue one from Eric Evans is highly encouraged ». Yes, of course the Blue Book is the reference book. But I would not necessarily advise to read that one first now that we have other books available that are not that hard to digest. Vladik’s book is a good one to start with in my opinion.

First things first: I’d add that if you don’t have access to domain experts, you cannot do this. Also I would emphasize even more that DDD is a learning process.

Define: « Senior developers or architects should spend time creating a Bounded Context Canvas to define clear boundaries. This helps formalize your understanding and link it to upcoming User Stories. Another helpful step is to create C4 diagrams and share them with developers. » This should be a team activity if you’re looking for alignment.

The Robot Vacuum metaphor and the diagram is gold!

**Comment by geovanneb** [comment]
Thank you very much for your valuable feedback @Max-Git 

The things that you pointed out really makes sense for me, so I have reflected [these changes](https://github.com/geovanneb/ddd-at-scale/commit/108f5da8b28312e2395c1820eaf65773b751ed18)

I hope it can be useful for other people in the future.

Regards!

**Reply by Max-Git** [reply, marked as answer]
Glad to read that my comments were useful. I like the changes you did. (There's a typo line 68: choaches)
